src/test/java/com/stormeye/evaluation/DeployGeneratedKeys.java

```diff
@@ -0,0 +1,180 @@
+import com.casper.sdk.identifier.block.HashBlockIdentifier;
+import com.casper.sdk.model.block.JsonBlockData;
+import com.casper.sdk.model.common.Digest;
+import com.casper.sdk.model.common.Ttl;
+import com.casper.sdk.model.deploy.Deploy;
+import com.casper.sdk.model.event.blockadded.BlockAdded;
+import com.casper.sdk.model.key.PublicKey;
+import com.casper.sdk.service.CasperService;
+import com.stormeye.utils.AssetUtils;
+import com.stormeye.utils.CasperClientProvider;
+import com.stormeye.utils.ContextMap;
+import com.syntifi.crypto.key.AbstractPrivateKey;
+import com.syntifi.crypto.key.AbstractPublicKey;
+import com.syntifi.crypto.key.Ed25519PrivateKey;
+import com.syntifi.crypto.key.Secp256k1PrivateKey;
+import dev.oak3.sbs4j.exception.ValueSerializationException;
+import io.cucumber.java.BeforeAll;
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.math.BigInteger;
+import java.net.URL;
+import java.security.GeneralSecurityException;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.Locale;
+import java.util.Random;
+
+import static com.stormeye.evaluation.StepConstants.*;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+import static org.hamcrest.core.IsNull.notNullValue;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * Step Definitions for Deploys with generated keys
+ */
+public class DeployGeneratedKeys {
+
+    private final ContextMap contextMap = ContextMap.getInstance();
+    private final Logger logger = LoggerFactory.getLogger(DeployGeneratedKeys.class);
+
+    @BeforeAll
+    public static void setUp() {
+        ContextMap.getInstance().clear();
+    }
+
+
+    @Given("that a {string} sender key is generated")
+    public void thatSenderKeyIsGenerated(final String algo) throws IOException, GeneralSecurityException {
+
+        logger.info("that a {} sender key is generated", algo);
+
+        final AbstractPrivateKey sk;
+        final AbstractPublicKey pk;
+
+        if (algo.equals("Ed25519")){
+            sk = CasperKeyHelper.createRandomEd25519Key();
+            pk = CasperKeyHelper.derivePublicKey((Ed25519PrivateKey) sk);
+        } else {
+            sk = CasperKeyHelper.createRandomSecp256k1Key();
+            pk = CasperKeyHelper.derivePublicKey((Secp256k1PrivateKey) sk);
+        }
+
+        assertThat(sk, is(notNullValue()));
+        assertThat(pk, is(notNullValue()));
+
+        byte[] msg = "this is the sender".getBytes();
+        byte[] signature = sk.sign(msg);
+        assertTrue(pk.verify(msg, signature));
+
+        assertThat(sk.getKey(), is(notNullValue()));
+        assertThat(pk.getKey(), is(notNullValue()));
+
+        contextMap.put(SENDER_KEY_SK, sk);
+        contextMap.put(SENDER_KEY_PK, pk);
+
+    }
+
+    @Then("fund the account from the faucet user")
```

Comment:
For consitancy across SDKs please provide the transfer amount and payment amount as step paramteres